### PR TITLE
metis: add support for conan v2

### PR DESCRIPTION
**Pull request by joakimono**
Specify library name and version:  **metis/5.1.1**

- Add support for conan v2
- Add new option `with_64bit_types` to allow choosing int and float types to be 32 or 64 bit. 32-bit types are required for some downstream packages
- Depends on gklib, with PR #17570 

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/metis/all/conanfile.py
Comment (SpaceIm) [inline recipes/metis/all/conanfile.py:91]:
```suggestion
```

File: recipes/metis/all/conanfile.py
Comment (SpaceIm) [inline recipes/metis/all/conanfile.py:103]:
```suggestion
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Comment by SpaceIm** [comment]
"Add support for conan v2" is not very informative given that there are 1500 recipes in conancenter, could you change this title please?

File: recipes/metis/all/conanfile.py
Comment (RubenRBS) [inline recipes/metis/all/conanfile.py:60]: No need to set this, already done by the CMakeToolchain integration :)

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Thanks a lot! Requesting changes in this PR so it does not get merged by accident while we wait for the other PR to get merged first, but otherwise the changes look good!

File: recipes/metis/all/conanfile.py
Comment (SpaceIm) [reply recipes/metis/all/conanfile.py:60]: It's required when cmake_minimum_required is less than 3.0, if you want to set install_name of shared lib to `@rpath` on Apple (relocatable shared lib). CMakeToolchain does not manage this policy.

Comment (SpaceIm) [reply recipes/metis/all/conanfile.py:60]: It would be enough, didn't see it. What is the rational behind this bump of cmake min version? It's not explained.

File: recipes/metis/all/conandata.yml
Comment (franramirez688) [inline recipes/metis/all/conandata.yml:22]: @joakimono why is this needed? I saw upstream is requiring 2.8, is it related to using the `CMAKE_POLICY_DEFAULT_CMP0042` policy?

**Review by franramirez688 — CHANGES_REQUESTED** [review]
Thanks for this contribution! Only one clarification before moving forward with the PR.

File: recipes/metis/all/conandata.yml
Comment (joakimono) [reply recipes/metis/all/conandata.yml:22]: I changed it due to remove another policy warning CMP0054 regarding quoted variables. I tried to suppress this like is done  with CMP0042 above, but for some reason it had no effect. The warning is triggered by the generated `cmakedeps_macros.cmake:33`. I assumed that changing the minimum required cmake in the upstream did not matter much, since the recipe requires 3.15 anyway. It should be noted that the metis CMakeLists.txt files seem pretty standard to me and do not use any deprecated 2.8 functionality.

**Comment by franramirez688** [comment]
It looked like it got stuck at some point. Updating the branch to see if the CI is passing for Conan v2 and v1.

**Comment by joakimono** [comment]
> It looked like it got stuck at some point. Updating the branch to see if the CI is passing for Conan v2 and v1.

The recently merged dependency gklib seems not to be built with the same build matrix configuration as this PR. Even though gklib was all green, it seems e.g. only apple-clang 13 was built (conan v1). #17570.

**Comment by SSE4** [comment]
@RubenRBS @franramirez688 can you unblock review?

**Comment by SSE4** [comment]
@franramirez688 @RubenRBS can you unblock? it seems like your questions were answered long time ago.

**Review by RubenRBS — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**
